### PR TITLE
introducing an easy-to-read naming scheme for our protocols

### PR DESCRIPTION
**Pull request by da-kami**
We have another HALight, ahm Highlight for you...

And yes, we are aware that the ticket said we should not create a spike. But creating an RFC right away seems a bit too fast 😉

### CODE REVIEW

File: 0021-protocol-naming.adoc [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+NOTE: Author: {authors} +
+Date: {revdate} +
+Tracking issue: https://github.com/comit-network/RFCs/issues/130[#130]
+
+== Proposal
+
+We propose the following protocol names for Alpha and Beta ledger:
+
+- `HAN` - HTCL for Asset on Native
```
Comment (tcharding) [inline 0021-protocol-naming.adoc]: Perhaps, 'HTLC Native Asset' - HNA

File: 0021-protocol-naming.adoc [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+- `HErc20` - HTLC for ERC20
+- `HOmni` - HTLC for OMNI
+- `HALigth` - HTLC for Lightning
```
Comment (tcharding) [inline 0021-protocol-naming.adoc]: Is this meant to be 'HTLC for Asset on Lighting'?  Perhaps 'HAL'?

File: 0021-protocol-naming.adoc (hunk 1 above)
Comment (tcharding) [reply 0021-protocol-naming.adoc]: The tokens would become 'HTLC Custom Asset' - HCA

Comment (D4nte) [reply 0021-protocol-naming.adoc]: Star Wars theme now?

Comment (D4nte) [reply 0021-protocol-naming.adoc]: I agree with @tcharding the `HTLC for Asset on Native` does not sound English.
`HTLC for Native Asset` sounds fine.

File: 0021-protocol-naming.adoc (hunk 2 above)
Comment (da-kami) [reply 0021-protocol-naming.adoc]: we had `HAL` in the beginning, but that is limiting if you assume that another special implementation for a ledger starting with `L` could exist. For specific implementation names we opted for longer names.

File: 0021-protocol-naming.adoc (hunk 1 above)
Comment (da-kami) [reply 0021-protocol-naming.adoc]: `HAN` just reads and sounds nicer, that's why we changed the order. In the beginning we had `NAH` (which is like a fun version of "no" in Austrian dialect). HAN just makes it easy to pronounce.

File: 0021-protocol-naming.adoc (hunk 2 above)
Comment (da-kami) [inline 0021-protocol-naming.adoc]:
```suggestion
- `HALight` - HTLC for Lightning
```

File: 0021-protocol-naming.adoc [hunk 3]
```diff
@@ -0,0 +1,46 @@
+= Introducing a protocol naming scheme
+Thomas Eizinger <thomas@coblox.tech>; Daniel Karzel<daniel.karzel@coblox.tech>
```
Comment (thomaseizinger) [inline 0021-protocol-naming.adoc]:
```suggestion
Thomas Eizinger <thomas@coblox.tech>; Daniel Karzel <daniel.karzel@coblox.tech>;
```

**Review by thomaseizinger — APPROVED** [review]
Awesome :)

File: 0021-protocol-naming.adoc (hunk 1 above)
Comment (D4nte) [reply 0021-protocol-naming.adoc]: Nonetheless "HTLC for asset on native" sounds weird.

Comment (thomaseizinger) [reply 0021-protocol-naming.adoc]: Agree with @da-kami here. The abbreviation is massively more important than how you expand it. The expansion of `HAN` could also read:

(Atomic swap using) **H**tlcs of **A**ssets that are **N**ative to the ledger

HAN can be easily pronounced, is short and contains all the important information to differentiate it from other protocols.

At this stage, I would suggest for reviewers to make a counter proposal that is at least equally good (measured by all the aspects mentioned in the spike) if they are not happy with the result.

File: 0021-protocol-naming.adoc (hunk 2 above)
Comment (tcharding) [reply 0021-protocol-naming.adoc]: Reasonable point.

**Comment by thomaseizinger** [comment]
@da-kami I pushed a commit that slightly revises the expansion of the abbreviations.

**Review by D4nte — COMMENTED** [review]
I have reviewed and provided my input.

**Review by luckysori — COMMENTED** [review]
Open question: what is the word asset (and the letter A) adding if everything that we deal with _is_ an asset?

**Comment by thomaseizinger** [comment]
@luckysori That is a good question! First thoughts I have on these are:
We could at some point have protocols that don't deal with assets? Opening channels maybe?

It doesn't hurt and makes the acronym sound better 🤷‍♂️

@d4nte, are you happier with the update of the expanded text?

**Review by D4nte — APPROVED** [review]
LGreatTM

File: 0021-protocol-naming.adoc (hunk 1 above)
Comment (tcharding) [reply 0021-protocol-naming.adoc]: > HAN just reads and sounds nicer, that's why we changed the order.
Don't tell fibs, this is a reference to the Chinese Han dynasty, you slipped it in hoping we wouldn't notice :)

Comment (tcharding) [reply 0021-protocol-naming.adoc]: I agree with @D4nte that "HTLC for asset on native" sounds rubbish but also with the others that 'HAN' is a better acronym.  "HTLC Asset Native" is fine though and satisfies both camps :)

Comment (D4nte) [reply 0021-protocol-naming.adoc]: You are late to the party they already changed the proposal ;)

**Comment by tcharding** [comment]
@da-kami MERGE yo!

**Pull request merged**